common/src/java/com/zimbra/common/soap/W3cDomUtil.java

```diff
@@ -91,29 +80,58 @@ private W3cDomUtil() {}
             }
     };
 
+    public static DocumentBuilderFactory makeDocumentBuilderFactory() throws ParserConfigurationException {
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        dbf.setNamespaceAware(true);
+        dbf.setIgnoringComments(true);
+        // protect against recursive entity expansion DOS attack and perhaps other things
+        dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        try {
+            // XXE attack prevention
+            dbf.setFeature(Constants.DISALLOW_DOCTYPE_DECL, true);
+            dbf.setFeature(Constants.EXTERNAL_GENERAL_ENTITIES, false);
+            dbf.setFeature(Constants.EXTERNAL_PARAMETER_ENTITIES, false);
+            dbf.setFeature(Constants.LOAD_EXTERNAL_DTD, false);
+            dbf.setXIncludeAware(false);
+            dbf.setExpandEntityReferences(false);
+        } catch (IllegalArgumentException iae) {
+            ZimbraLog.misc.debug("Disabling doctype-decl not supported", iae);
```

Comment:
(Sorry for the late review) A few questions here...
1. Should we allow the disabling of doctype-decl failure to fall through and return dbf still?  That seems ill advised, but perhaps I'm missing something.
2. In addition, is the IllegalArgumentException specific to doctype-decl only? I'm guessing no and perhaps it's more just any setFeature or setXIncludeAware request which can throw this, and we didn't update the debug message appropriately.
3. Does the order of lines 91-95 matter?  Per Q1 above, If we're allowing a failure to successfully follow through, shouldn't we try each setFeature in a separate try/catch block in case only one of those attempts fails but the others can still succeed? Or is the assumption if one fails, they will all fail?